src/main/java/eu/tornplayground/tornapi/mappers/UserMapper.java

```diff
@@ -91,17 +96,31 @@ public static List<Honor> ofHonors(JsonNode json) {
 
         List<Honor> honors = new ArrayList<>();
         for (int i = 0; i < awarded.size(); i++) {
-            Honor honor = new Honor();
-            honor.setId(awarded.get(i));
-            honor.setAwarded(times.get(i));
+            try {
+                Constructor<Honor> honorConstructor = Honor.class.getDeclaredConstructor();
+                honorConstructor.setAccessible(true);
 
-            honors.add(honor);
+                Honor honor = honorConstructor.newInstance();
+
+                Field idField = Honor.class.getDeclaredField("id");
+                idField.setAccessible(true);
+                idField.set(honor, awarded.get(i));
+
+                Field awardedField = Honor.class.getDeclaredField("awarded");
+                awardedField.setAccessible(true);
+                awardedField.set(honor, times.get(i));
+
+                honors.add(honor);
+            } catch (NoSuchMethodException | NoSuchFieldException | InvocationTargetException | InstantiationException |
+                     IllegalAccessException e) {
+                throw new RuntimeException(e);
+            }
```

Comment:
Reflection here seems stupid, just do an `@AllArgsConstructor` on the model and create the object from the constructor.